src/Elasticsearch.Net/Transport/Pipeline/RequestData.cs

```diff
@@ -97,25 +105,26 @@ private string CreatePathWithQueryStrings(string path, IConnectionConfigurationV
 			var tempUri = new Uri("http://localhost:9200/" + path).Purify();
 			if (tempUri.Query.IsNullOrEmpty())
 				path += queryString;
-			else 
+			else
 				path += "&" + queryString.Substring(1, queryString.Length - 1);
 			return path;
 		}
 
-		protected bool Equals(RequestData other) => 
-			RequestTimeout.Equals(other.RequestTimeout) 
-			&& PingTimeout.Equals(other.PingTimeout) 
-			&& KeepAliveTime == other.KeepAliveTime 
-			&& KeepAliveInterval == other.KeepAliveInterval 
-			&& Pipelined == other.Pipelined 
-			&& HttpCompression == other.HttpCompression 
-			&& Equals(Headers, other.Headers) 
-			&& string.Equals(ProxyAddress, other.ProxyAddress) 
-			&& string.Equals(ProxyUsername, other.ProxyUsername) 
-			&& string.Equals(ProxyPassword, other.ProxyPassword) 
-			&& DisableAutomaticProxyDetection == other.DisableAutomaticProxyDetection 
-			&& Equals(BasicAuthorizationCredentials, other.BasicAuthorizationCredentials) 
-			&& Equals(ConnectionSettings, other.ConnectionSettings) 
+		protected bool Equals(RequestData other) =>
+			RequestTimeout.Equals(other.RequestTimeout)
+			&& PingTimeout.Equals(other.PingTimeout)
+			&& KeepAliveTime == other.KeepAliveTime
+			&& KeepAliveInterval == other.KeepAliveInterval
+			&& Pipelined == other.Pipelined
+			&& HttpCompression == other.HttpCompression
+			&& Equals(Headers, other.Headers)
+			&& string.Equals(RunAs, other.RunAs)
+			&& string.Equals(ProxyAddress, other.ProxyAddress)
+			&& string.Equals(ProxyUsername, other.ProxyUsername)
+			&& string.Equals(ProxyPassword, other.ProxyPassword)
+			&& DisableAutomaticProxyDetection == other.DisableAutomaticProxyDetection
+			&& Equals(BasicAuthorizationCredentials, other.BasicAuthorizationCredentials)

```

Comment:
Not really related to this PR, but we should mark this obsolete and proxy `BasicAuthenticationCredentials`.